habitat_baselines/il/data/data.py

```diff
@@ -0,0 +1,255 @@
+    ):
+        """
+        Args:
+            config: Config
+            input_type (string): Type of model being trained ("vqa", "pacman")
+            num_frames (int): number of frames used as input to VQA model
+            max_controller_actions (int):
+        """
+        self.config = config.TASK_CONFIG
+        self.env = habitat.Env(config=self.config)
+        self.input_type = input_type
+        self.num_frames = num_frames
+
+        self.episodes = self.env._dataset.episodes
+
+        # sorting and making episode ids consecutive for simpler indexing
+        self.sort_episodes()
+
+        self.q_vocab = self.env._dataset.question_vocab
+        self.ans_vocab = self.env._dataset.answer_vocab
+
+        self.eval_save_results = config.EVAL_SAVE_RESULTS
+
+        if self.config.DATASET.SPLIT == config.EVAL.SPLIT:
+            self.mode = "val"
+        else:
+            self.mode = "train"
+
+        self.frame_dataset_path = config.FRAME_DATASET_PATH.format(
+            split=self.mode
+        )
+
+        group_by_keys = filters.Curried(self.group_by_keys_)
+        super().__init__(
+            urls=self.frame_dataset_path + ".tar",
+            initial_pipeline=[group_by_keys()],
+        )
+
+        self.only_vqa_task = config.ONLY_VQA_TASK
+
+        # [TODO] can be done in mp3d_eqa_dataset while loading dataset
+        self.calc_max_length()
+        self.restructure_ans_vocab()
+
+        self.scene_episode_dict = get_scene_episode_dict(self.episodes)
+
+        if not self.cache_exists():
+            """
+            for each scene > load scene in memory > save frames for each
+            episode corresponding to each scene
+            """
+            logger.info(
+                "[ Dataset cache not present / is incomplete. ]\
+                \n[ Saving episode frames to disk. ]"
+            )
+
+            logger.info(
+                "Number of {} episodes: {}".format(
+                    self.mode, len(self.episodes)
+                )
+            )
+
+            for scene in tqdm(
+                list(self.scene_episode_dict.keys()),
+                desc="Going through all scenes from dataset",
+            ):
+
+                self.load_scene(scene)
+
+                for episode in tqdm(
+                    self.scene_episode_dict[scene],
+                    desc="Saving episode frames for each scene",
+                ):
+
+                    if self.only_vqa_task:
+                        pos_queue = episode.shortest_paths[0][
+                            -self.num_frames :  # noqa: E203
+                        ]
+                    else:
+                        pos_queue = episode.shortest_paths[0]
+
+                    self.save_frame_queue(pos_queue, episode.episode_id)
+
+            logger.info("[ Saved all episodes' frames to disk. ]")
+
+            create_tar_archive(
+                self.frame_dataset_path + ".tar", self.frame_dataset_path
+            )
```

Comment:
I think we should remove the contents of `self.frame_dataset_path` after making the tarball?